stubs/common/Helpers.stub

```diff
@@ -189,3 +189,14 @@ function filled($value) {}
  * @return bool
  */
 function blank($value) {}
+
+/**
+ * Gets the value of an environment variable.
+ *
+ * @template TDefault
+ *
+ * @param  string  $key
+ * @param  TDefault  $default
+ * @return ($default is null ? mixed : TDefault)
```

Comment:
What about the following?

```suggestion
 * @return bool|string|TDefault
```

This would yield the following types:

```php
assertType('bool|string|null', env('foo'));
assertType('bool|string|null', env('foo', null));
assertType('bool|string', env('foo', ''));
assertType('bool|int|string', env('foo', 0));
```

Reply:
Looking at the internals of `env()`, you seem to be correct.

That's even cleaner indeed. I will apply your suggestion.
Sorry for the back and forth, I am not too familiar yet contributing to PHPStan.